src/ring_jwt_middleware/core.clj

```diff
@@ -233,68 +276,85 @@
+                (->pure {:identity (post-jwt-format-fn jwt)
+                         :jwt jwt}))]
+          (handler (into request (<-result authentication-result))))))))
+
+(defn mk-wrap-authorization
+  "A function building a middleware taking care of the authorization logic.
+
+  It must be used in conjunction with `mk-wrap-authentication`.
+
+  The configuration is map containing two handlers.
+
+  - no-jwt-handler => a function that will take a handler and return a new web handler.
+                      This will be used when no JWT is present.
+                      As the function takes the handler, you can still continue without blocking the access.
+  - error-handler => a function taking a JWT error (see Result) and returning a ring response.
+                     This function should generally just return a 401 (unauthorized)."
+  [{:keys [no-jwt-handler
+           error-handler]
+    :or {no-jwt-handler forbid-no-jwt-header-strategy}}]
+  (let [handle-error (or error-handler default-error-handler)]
     (fn [handler]
       (let [no-jwt-fn (no-jwt-handler handler)]
-        (fn [request]
-          (if-let [raw-jwt (get-jwt request)]
-            (if-let [jwt (decode raw-jwt p-fn structured-log-fn)]
-              (if-let [validation-errors
-                       (validate-jwt raw-jwt
-                                     jwt
-                                     jwt-max-lifetime-in-sec
-                                     jwt-check-fn
-                                     structured-log-fn)]
-                (handle-error (format "(%s) %s"
-                                      (or (jwt->user-id jwt) "Unknown User ID")
-                                      (str/join ", " validation-errors))
-                              {:jwt jwt
-                               :error :jwt_validation_error
-                               :errors validation-errors})
-                (if (try (is-revoked-fn jwt)
-                         (catch Exception e
-                           (structured-log-fn "is-revoked-fn thrown an exception for"
-                                              {:level :error
-                                               :jwt jwt})
-                           (throw e)))
-                  (handle-error (format "JWT revoked for %s"
-                                        (or (jwt->user-id jwt) "Unknown User ID"))
-                                {:jwt jwt
-                                 :error :jwt_revoked})
-                  (handler (assoc request
-                                  :identity (post-jwt-format-fn jwt)
-                                  :jwt jwt))))
-              (handle-error "Invalid Authorization Header (couldn't verify the JWT signature)"
-                            {:authorization-header (str "Bearer:" raw-jwt)}))
-            (no-jwt-fn request)))))))
+        (fn [{{:keys [error] :as jwt-error} :jwt-error
+              :as request}]
+          (if jwt-error
```

Comment:
Instead of checking the `jwt-error` I would check the presence of the `identity` or the `jwt` key to know if the request is authenticated. Not having a `jwt-error` doesn't mean that the request is authenticated.

You could extract an `authenticated?` function to better expose the logic in this middleware. I saw something similar in [buddy-auth](https://github.com/funcool/buddy-auth/blob/f8ff8693ed2b328e913a77806e20ebd8142a1dbe/src/buddy/auth.clj#L27)